### ISSUE TITLE
Improve exception reporting

### ISSUE THREAD

**Issue by jreese** [facebookincubator/Bowler#63]
When transforms or modifiers fail, Bowler prints an error, and stringifies the exception, but this doesn't always help understand what the actual problem is (sometimes it just prints the Node, which is useless).  Maybe print a stack trace, or at the very list, print the type of the exception along with the stringified output.

**Comment by DanielMarchand** [comment]
Related to this issue (though not as complete) would like to improve the logging for one of the Exceptions:

Bowler/bowler/tool.py: 157/158

except ParseError as e:
     log.error(f"Skipping {filename}: failed to parse because {e}")

**Comment by thatch** [comment]
Yeah, at least in debug mode this would be nice.  There's code in https://github.com/facebookincubator/Bowler/blob/master/bowler/tests/lib.py#L75 and L108 and L149 to make sure we can see the full stack in tests (I'm a test-first kind of person, so this was enough for me).

**Comment by thatch** [comment]
@jreese do you remember why ParseError is more silent, rather than falling through?  This doesn't go through the code path improved by https://github.com/facebookincubator/Bowler/pull/86

**Comment by jreese** [comment]
I believe the reason was to prevent aborting the entire query early in cases of hitting some malformed (or py2/print statement) files.  Ie, when running a particularly wide or expensive query, better to allow the query to run to completion on all possible files than to abort and potentially force the dev to repeat the expensive query.  Ideally, we should probably gather these errors and report them at the end, or at least return a failure code of some sort.  I'm open to other ideas or opinions.
